nerfstudio/models/tensorf.py

```diff
@@ -0,0 +1,282 @@
+from nerfstudio.model_components.ray_samplers import PDFSampler, UniformSampler
+from nerfstudio.model_components.renderers import (
+    AccumulationRenderer,
+    DepthRenderer,
+    RGBRenderer,
+)
+from nerfstudio.model_components.scene_colliders import AABBBoxCollider
+from nerfstudio.models.base_model import Model, VanillaModelConfig
+from nerfstudio.utils import colormaps, colors, misc
+
+
+@dataclass
+class TensoRFModelConfig(VanillaModelConfig):
+    """TensoRF model config"""
+
+    _target: Type = field(default_factory=lambda: TensoRFModel)
+    """target class to instantiate"""
+    init_resolution: int = 128
+    """initial render resolution"""
+    final_resolution: int = 300
+    """final render resolution"""
+    upsampling_iters: Tuple[int, ...] = (2000, 3000, 4000, 5500, 7000)
+    """specifies a list of iteration step numbers to perform upsampling"""
+    loss_coefficients: Dict[str, float] = to_immutable_dict({"rgb_loss": 1.0})
+    """Loss specific weights."""
+    num_samples: int = 256
+    """Number of samples in field evaluation"""
+    num_den_components: int = 16
+    """Number of components in density encoding"""
+    num_color_components: int = 48
+    """Number of components in color encoding"""
+    appearance_dim: int = 27
+    """Number of channels for color encoding"""
+
+
+class TensoRFModel(Model):
+    """TensoRF Model
+
+    Args:
+        config: TensoRF configuration to instantiate model
+    """
+
+    def __init__(
+        self,
+        config: TensoRFModelConfig,
+        **kwargs,
+    ) -> None:
+        self.init_resolution = config.init_resolution
+        self.upsampling_iters = config.upsampling_iters
+        self.num_den_components = config.num_den_components
+        self.num_color_components = config.num_color_components
+        self.appearance_dim = config.appearance_dim
+        self.upsampling_steps = (
+            np.round(
+                np.exp(
+                    np.linspace(
+                        np.log(config.init_resolution),
+                        np.log(config.final_resolution),
+                        len(config.upsampling_iters) + 1,
+                    )
+                )
+            )
+            .astype("int")
+            .tolist()[1:]
+        )
+        super().__init__(config=config, **kwargs)
+
+    def get_training_callbacks(
+        self, training_callback_attributes: TrainingCallbackAttributes
+    ) -> List[TrainingCallback]:
+
+        # the callback that we want to run every X iterations after the training iteration
+        def reinitialize_optimizer(
+            self, training_callback_attributes: TrainingCallbackAttributes, step: int  # pylint: disable=unused-argument
+        ):
+            resolution = self.upsampling_steps.pop(0)
+
+            # upsample the position and direction grids
+            # TODO(ethan): ask Brent how to get typing to work on this... the Encoding base class type
+            # in NeRFField is causing the issue
```

Comment:
Is this still an issue?

Reply:
I think this should be fine with the new TensoRFField